src/pydsl/triton.py

```diff
@@ -0,0 +1,238 @@
+
+    # Remove unused arguments from the entry block
+    for idx in indices_to_remove:
+        entry_block.erase_argument(idx)
+
+    # Update function type
+    old_func_type = func_op.type
+    new_inputs = [
+        typ
+        for i, typ in enumerate(old_func_type.inputs)
+        if i not in indices_to_remove
+    ]
+    new_func_type = mlirir.FunctionType.get(new_inputs, old_func_type.results)
+    func_op.attributes["function_type"] = mlirir.TypeAttr.get(new_func_type)
+
+
+def combine_mlir_modules(
+    main_module: mlirir.Module,
+    module2: mlirir.Module,
+    triton_funcs: dict,
+    current_signature: tuple[str, ...],
+) -> func.FuncOp:
+    """
+    Combines two MLIR modules by copying the operations from `module2` into
+    `main_module`. Checks to see if the function has already been created, and
+    if it has, checks to see if the signature has been created. If the function
+    exists with a different signature, add "n" to the function name, where "n"
+    is the current number of distinct signatures.
+    """
+    with InsertionPoint(main_module.body):
+        prefix = "__triton_to_pydsl_"
+        callees = []
+        for op in module2.body.operations:
+            if isinstance(op, func.FuncOp):
+                signatures = triton_funcs.setdefault(op.sym_name.value, {})
+                callee_name = (
+                    op.sym_name.value + prefix + str(len(signatures))
+                )  # add a number to the end of the function name
+                if current_signature in signatures.keys():
+                    callees.append(signatures[current_signature])
+                    continue  # this function has already been compiled
+                op.attributes["sym_name"] = mlirir.StringAttr.get(callee_name)
+                new_op = op.clone()
+                remove_unnecessary_args(new_op)
+                signatures[current_signature] = new_op
+                callees.append(new_op)
+            else:
+                op.clone()
+    assert len(callees) == 1  # make sure there is only one FuncOp in module2
+    return callees[0]
+
+
+def run_and_get_output_stdin(
+    cmds: Iterable[str], stdin_string: str
+) -> str | None:
+    result = subprocess.run(
+        " ".join(cmds),
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        input=stdin_string.encode("utf-8"),
+        shell=True,
+        check=True,
+    )
+    return result.stdout.decode("utf-8") if result.stdout else None
+
+
+def triton_to_linalg(
+    module: mlirir.Module,
+    triton_func: JITFunction,
+    args: list[tuple[str, typing.Any]],
+    triton_funcs: list,
+) -> func.FuncOp:
+    """
+    Converts triton functions to linalg dialect using triton-adapter-opt.
+    args is a list of tuples of the form (name, value), where name is the
+    name of the triton function argument and type is the PyDSL parameter that
+    the function has been called with.
+    Returns the converted function as a func.FuncOp.
+    """
```

Comment:
```suggestion
    """
    Converts triton functions to linalg dialect using triton-adapter-opt.
    args is a list of tuples of the form (name, value), where name is the
    name of the triton function argument and value is the PyDSL argument that
    the function has been called with.
    Returns the converted function as a func.FuncOp.
    """
```
Forgot to update this type -> value after renaming yesterday.